lib/handlers/quote/quote.ts

```diff
@@ -642,6 +624,50 @@ export class QuoteHandler extends APIGLambdaHandler<
     }
   }
 
+  static protocolsFromRequest(
+    chainId: ChainId,
+    requestedProtocols: string[] | string | undefined,
+    requestSource: string,
+    appVersion: string | undefined,
+    forceCrossProtocol: boolean | undefined
+  ): Protocol[] {
+    const isMobileRequest = ['uniswap-ios', 'uniswap-android'].includes(requestSource)
+    // We will exclude V2 if isMobile and the appVersion is not present or is lower than 1.24
+    const semverAppVersion = semver.coerce(appVersion)
+    const fixVersion = semver.coerce('1.24')!
+    const excludeV2 = isMobileRequest && (semverAppVersion === null || semver.lt(semverAppVersion, fixVersion))
+
+    if (requestedProtocols) {
+      let protocols: Protocol[] = []
+
+      for (const protocolStr of requestedProtocols) {
+        switch (protocolStr.toUpperCase()) {
+          case Protocol.V2:
+            if (chainId === ChainId.MAINNET || !excludeV2) {
+              protocols.push(Protocol.V2)
+            }
+            break
+          case Protocol.V3:
+            protocols.push(Protocol.V3)
+            break
+          case Protocol.MIXED:
+            if (chainId === ChainId.MAINNET || !excludeV2) {
+              protocols.push(Protocol.MIXED)
+            }
+            break
+          default:
+            return []
+        }
+      }
+
+      return protocols
+    } else if (!forceCrossProtocol) {
```

Comment:
@mikeki why is this line v3 only?

Reply:
thats a good question, I'm actually not sure, it is probably because in the past we didnt have v2 support on all chains and we were default to v3 only. This is how it was before though, so I didnt change the business logic there